src/Framework/Properties/AssemblyInfo.cs

```diff
@@ -45,6 +45,7 @@
 [assembly: InternalsVisibleTo("MSBuild, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
 [assembly: InternalsVisibleTo("Microsoft.Build.Conversion.Core, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
 
+[assembly: InternalsVisibleTo("Microsoft.Build.Engine.OM.UnitTests, PublicKey=002400000480000094000000060200000024000052534131000400000100010015c01ae1f50e8cc09ba9eac9147cf8fd9fce2cfe9f8dce4f7301c4132ca9fb50ce8cbf1df4dc18dd4d210e4345c744ecb3365ed327efdbc52603faa5e21daa11234c8c4a73e51f03bf192544581ebe107adee3a34928e39d04e524a9ce729d5090bfd7dad9d10c722c0def9ccc08ff0a03790e48bcd1f9b6c476063e1966a1c4")]
```

Comment:
This was required to be able to reset change wave state from `TestEnvironment`. I can move the `SetChangeWave` function out of `TestEnvironment` and keep it all in `ChangeWaves_Tests`, we'd wouldn't be able to "nicely" clean up change waves as `TestEnvironment` disposes.

Reply:
Notes as I'm doing this change: This makes it slightly messier to reset change waves before each test uses it. But if it prevents IVT on a project that was intended not to have it, it's a small price to pay & small doc update.

Tests go from `env.SetChangeWave` to 
```
            ChangeWaves.ResetStateForTests();
            env.SetEnvironmentVariable("MSBUILDDISABLEFEATURESFROMVERSION", wave);
            BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
```

Reply:
Tests also need to reset changewaves once the test is complete, to prevent tests that are affected by the new behavior (ChangeWaves are opt-**out**, not in).

Say `Test A` runs and set changewave to 16.8 to test a change in behavior. `Test B` (from the same assembly) will see changewave set to 16.8, and instead of running its existing behavior to test it, it will go down the ChangeWave code path.

So every test that calls changewaves will need to call 
```
ChangeWaves.ResetStateForTests();
env.SetEnvironmentVariable("MSBUILDDISABLEFEATURESFROMVERSION", TheirWave);
BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
```
At the beginning and end of each test, and the end call should pass `String.Empty`

Reply:
hmm, I definitely prefer the `env.SetChangeWave` approach. Is the problem that resetting is not accessible to the environment itself? Maybe we should finally go through with splitting out a TestHelpers assembly that can have IVT?

Reply:
So after touching this up a bit, the only thing future CHangeWaves tests have to do is:
```
ChangeWaves.ResetStateForTests();
env.SetEnvironmentVariable("MSBUILDDISABLEFEATURESFROMVERSION", TheirWave);
BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();
```

And nothing at the end of the test. I forget TestEnvironment can still set the env var to empty (bypassing the variant) and call and `BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly();` at `Dispose` time .

Reply:
I wonder if `BuildEnvironmentHelper.ResetInstance_ForUnitTestsOnly()` even needs to be called when they reset state and set the environment variable?

Reply:
I lied. We still need to call `ChangeWaves.ResetStateForTests` at the end of each function that uses `ChangeWaves`. A perfect example is `ExpandAllIntoStringNotTruncated` and `ExpandAllIntoStringTruncated`. `ExpandAllIntoStringNotTruncated` tests functionality if users opt-out of that wave so it sets the ChangeWave to `16.8`. `ExpandAllIntoStringTruncated` runs after it and expects the opt-in behavior to happen, but it doesn't because the static ChangeWave was set and not reset.

So ultimately it looks like we need to call this at the end of each test _just in case_, but it's a small price to pay. I'm interested in your idea of splitting `TestHelpers` out, but I think that can be handled/tracked by a different issue/PR.

Reply:
> Is the problem that resetting is not accessible to the environment itself?
Yes, we can't reset state unless we have IVT here.